File: Programmers/LEVEL_1/c++/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int sum = 0;
    int num = x;
    while (num > 0)
    {
        int temp = num % 10;
        num /= 10;
        sum += temp;
    }

    if (x % sum != 0)
        answer = false;

    return answer;
}

File: Programmers/LEVEL_1/하샤드 수.js
function solution(x) {
  var answer = true;

  let hashad = 0;
  let temp = x;
  while (temp > 0) {
    hashad += temp % 10;
    temp = parseInt(temp / 10, 10);
  }

  if (x % hashad != 0) {
    answer = false;
  }

  return answer;
}
